utils/bitfield/src/lib.rs

```diff
@@ -0,0 +1,453 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+pub mod bitvec_serde;
+pub mod rleplus;
+pub use bitvec;
+
+use bitvec::prelude::Lsb0;
+use core::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, Not};
+use fnv::FnvHashSet;
+use std::iter::FromIterator;
+
+type BitVec = bitvec::prelude::BitVec<Lsb0, u8>;
+type Result<T> = std::result::Result<T, &'static str>;
+
+/// Represents a bitfield to track bits set at indexes in the range of `u64`.
+#[derive(Debug)]
+pub enum BitField {
+    Encoded {
+        bv: BitVec,
+        set: FnvHashSet<u64>,
+        unset: FnvHashSet<u64>,
+    },
+    // TODO would be beneficial in future to only keep encoded bitvec in memory, but comes at a cost
+    Decoded(BitVec),
+}
+
+impl Default for BitField {
+    fn default() -> Self {
+        Self::Decoded(BitVec::new())
+    }
+}
+
+impl BitField {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Generates a new bitfield with a slice of all indexes to set.
+    pub fn new_from_set(set_bits: &[u64]) -> Self {
```

Comment:
`from_set` kinda makes it sound like you pass in a set, maybe add `_bits`? Or implementing `FromIterator<u64>` is another option

Reply:
Yeah this just has a very specific use so I was specifically trying to avoid making it generic (my thought was making the param generic on IntoIter to be more ergonomic), but was specifically trying to match go impl here. If any other use is needed it can always be changed